### ISSUE TITLE
Error updating plugins

### ISSUE THREAD

**Issue by flostadler** [zeit/hyper#3595]
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: Windows 10, Version 1803 (OS Build 17134.706)
- **Hyper.app version**: 3.0.0
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**:
- **Relevant information from devtools** _(CMD+ALT+I on macOS, CTRL+SHIFT+I elsewhere)_:
- **The issue is reproducible in vanilla Hyper.app**: `Is Vanilla`

## Issue
During updating the plugins I get following error message:
![grafik](https://user-images.githubusercontent.com/2453580/57220198-b4f56900-6ffa-11e9-99cf-89a6c9774629.png)

**Comment by pcnate** [comment]
have you tried running `hyper version`? I had an issue (#3602) where the CLI was showing an old version while the running app shows 3.0.0

**Comment by jarckn** [comment]
> have you tried running `hyper version`? I had an issue (#3602) where the CLI was showing an old version while the running app shows 3.0.0

I have the issue as well and `hyper version` is showing the correct version:
![image](https://user-images.githubusercontent.com/1441097/57288312-812d4880-70b9-11e9-88b3-1d16c1397ed7.png)

**Comment by flostadler** [comment]
`hyper version` is showing `3.0.0`

**Comment by pcnate** [comment]
another suggestion is https://github.com/zeit/hyper/issues/3590#issuecomment-489672067

**Comment by jarckn** [comment]
> `hyper version` is showing `3.0.0`

Are you using Hyper behind a proxy? Not using a proxy fixed this for me.

**Comment by flostadler** [comment]
Nope, no proxy either

**Comment by VincentLanglet** [comment]
I have the issue aswell

**Comment by bogdancss** [comment]
same here, on MacOS Mojave with hyper v3.0.2

**Comment by gmcouto** [comment]
same here
Windows 10
Hyper 3.0.2
npm 6.1.0
node 10.6.0

**Comment by vannaka** [comment]
I'm having the same problem.
Windows 10
Hyper 3.0.2
npm 6.9.0
node 10.16.0

**Comment by zaneclaes** [comment]
Same. I get the annoying pop-up many times per day for weeks now.
OSX Mojave
Hyper 3.0.2
npm 6.9.0
node 12.3.0

**Comment by guyeu** [comment]
Same...
OSX Mojave 10.14.5
Hyper 3.0.2
npm 6.9.0
node v12.6.0

**Comment by zachbryant** [comment]
Same.
Ubuntu 18.04
Hyper 3.0.2
npm 6.9.0
node v12.6.0

**Comment by briantanseng** [comment]
Same.
OSX Mojave 10.14.6
Hyper 3.0.2
npm 6.9.0
node v10.16.0

**Comment by dylanbr0wn** [comment]
..Same here
OSX Mojave 10.14.6
Hyper 3.0.2
npm 6.10.2
node v12.9.1

**Comment by samu** [comment]
Please stop spamming this thread (it's annoying for people who have subscribed to this thread for meaningful updates). Use the 👍 emoji on the issue description.

**Comment by alvitawa** [comment]
You should run `hyper` from the command line and see if the full error pops up there since it seems that you can't see the full notification text on windows. For me it said `node` is not recognized as a command so I installed nodejs. The plugin I was trying to run still didn't work but maybe that was the plugin's fault.

**Comment by vivekrp** [comment]
Any fixes for this? I keep getting this error pop-up every time I open/re-open Hyper. Tried everything, removed all plugins, uninstalled Hyper and installed again, reinstalled node & npm. Still no luck. 

Error updating plugins.
[DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
![Screenshot 2019-10-10 at 6 22 55 PM](https://user-images.githubusercontent.com/171857/66570329-05b15500-eb8b-11e9-8a09-35760481d30f.png)

**Comment by Stanzilla** [comment]
can you try a CI build from canary?

**Comment by vivekrp** [comment]
> can you try a CI build from canary?

I am a bit confused, did you mean changing stable to the canary in .hyper.js and restarting? Tried that didn't work.

or

Cloning the new canary build from GitHub and running it locally?

**Comment by Stanzilla** [comment]
You're on macOS, right? Download the dmg from here then https://app.circleci.com/jobs/github/zeit/hyper/2433/artifacts

**Comment by Stanzilla** [comment]
okay can you open (or search, I think there is an issue about that error already) an issue about that please.

**Comment by IgorGanapolsky** [comment]
What is the solution?  I am getting the same warning pop up on Mac OS and Hyper 3.0.2...

**Comment by vannaka** [comment]
@IgorGanapolsky, did you try running the Hyper 3.1.0 canary build [here](https://app.circleci.com/jobs/github/zeit/hyper/2433/artifacts)? It should fix the "Error updating plugins problem" but leads to a new problem, [Issue 3670](https://github.com/zeit/hyper/issues/3670) "Cannot read property 'properties' of undefined".

**Comment by IgorGanapolsky** [comment]
@vannaka **3.1.0** doesn't work at all.  It is frozen and nothing's clickable:
<img width="827" alt="Screen Shot 2019-10-15 at 12 33 14 PM" src="https://user-images.githubusercontent.com/201209/66851062-fe51c900-ef47-11e9-8384-8cc9dda0910e.png">

**Comment by Stanzilla** [comment]
turn of your plugins please

**Comment by Stanzilla** [comment]
open your config and delete the plugins names from the array

```js
plugins: [],
```

**Comment by IgorGanapolsky** [comment]
@Stanzilla Which plugins are compatible?

**Comment by CanRau** [comment]
3.1.0 canary1 works for me so far, I'm on macOS Mojave 10.14.6 and I've got hyperline & hyper-search plugins successfully running 👍

**Comment by Ryan315** [comment]
hyperline will cause that warning.

**Comment by IgorGanapolsky** [comment]
@Ryan315. I disabled hyperline and still getting the warning...

**Comment by ashfaqnisar** [comment]
I had the same issue, just use the canary version until the issue is resolved

**Comment by kgleason** [comment]
Sorry to resurrect a closed issue, but I have this exact issue in Windows 10 on Hyper 3.0.2. The link for the canary build 404s, and I don't really see any other proposed resolution.

**Comment by IgorGanapolsky** [comment]
Yeah this is still an issue in the latest Hyperterm, regardless of the plugins.

**Comment by albertorb** [comment]
In MacOS I solved it by just installing npm with brew.

**Comment by andyedwards79** [comment]
Just to confirm, I tried most of the solutions above - instaling both npm and hyper via brew did it for me too.

**Comment by luixal** [comment]
I'm still getting this issue on win10 and hyper 3.0.2 :(

**Comment by mariadb-faisalsaeed** [comment]
Just to give you guys an update and hopefully, some of the folks here will find comfort in this crazy painful COVID-19 time :)

I have been getting the same problem, it all started when I copied someone's configuration to make my Hyper look cool and I ended up with this error, Hyper was not even opening. Uninstall/reinstall did not help.

Here is what worked for me on **Windows 10**:

1. Uninstall Hyper
2. Remove the following two folders, as these are not removed automatically. 
- `C:\Users\**<user>**\AppData\Roaming\Hyper`
- `C:\Users\**<user>**\AppData\Local\hyper`
3. Install the latest Hyper, for me it was `hyper-Setup-3.0.2.exe`

Send some prayers for the entire world if this worked for you :)

**Comment by kamtouboris** [comment]
> Just to give you guys an update and hopefully, some of the folks here will find comfort in this crazy painful COVID-19 time :)
> 
> I have been getting the same problem, it all started when I copied someone's configuration to make my Hyper look cool and I ended up with this error, Hyper was not even opening. Uninstall/reinstall did not help.
> 
> Here is what worked for me on **Windows 10**:
> 
> 1. Uninstall Hyper
> 2. Remove the following two folders, as these are not removed automatically.
> 
> * `C:\Users\<user>\AppData\Roaming\Hyper`
> * `C:\Users\<user>\AppData\Local\hyper`
> 
> 1. Install the latest Hyper, for me it was `hyper-Setup-3.0.2.exe`
> 
> **Note:** _remember to replace with your own Windows user :p_
> 
> Send some prayers for the entire world if this worked for you :)

Works for me. Thx.

**Comment by brcmesquita** [comment]
I had the same problem here.

The problem is complicated to discover, but it is more than obvious.

We install "hyper" on Windows, via Windows, and then we want to install plugins etc via WSL (Bash Linux). It will not work!

To resolve, go back to the default configuration of Hyper (deleting ".hyper.js" for example) where it will open with the "cmd" Windows again. Install all the plugins you want (with "hyper install hypercwd" for example), and only after that, you change to WSL.

Applause! kkkkkkkkkkk

**Comment by schulzf** [comment]
I had this issue (twice - once for my mac mini m1 and once for my macbook air m1)
For me it was a permissions problem. I noticed that because every time I had to do a npm install / yarn add I got EACCES permission denied, I had to sudo everything - turns out that this was also messing up the ability of hyper to install plugins as well.

try doing this:

chown -R yourusername /users/yourusername

that solved the above issues I was having on my m1 mac

**Comment by davidrock** [comment]
The solution for me was to clear the plugins list in the  .hyper.js file and removing the packages inside the .hyper_plugins  folder.

**Comment by JigarJoshi04** [comment]
For me, the issue solved by removing the plugin "hyperline".
I would suggest start removing one by one plugin and update.
Some plugin is creating this issue.

**Comment by skm7463** [comment]
Ah yes, I've tested so many plugins and most don't really work well when put under strain.

I'm actually developing my own plugin to do far more than anything I can find out there and to avoid the trap of having to have lots of small plugins to make to give features I need. I need UI enhancements and automation options.

The problem of using multiple plugins is also that few actually work together well. 

The freeze that I reported is testable without any plugins (raw hyper). It seems at first that it might have been issue for you but glad you solve it.
